cli/setup.sh

```diff
@@ -0,0 +1,211 @@
+}
+
+function options() {
+
+  # Parse options
+  while getopts ":hi:" option; do
+      case "${option}" in
+          i|p) TEMPLATE_ARGS="${TEMPLATE_ARGS} -${option} ${OPTARG}" ;;
+          h) usage; return 1 ;;
+          \?) fatalOption; return 1 ;;
+      esac
+  done
+
+  return 0
+}
+
+function usage() {
+  cat <<EOF
+
+DESCRIPTION:
+  sets up your hamlet tenant ready for output generation
+
+USAGE:
+  $(basename $0)
+
+PARAMETERS:
+
+    -h                         shows this text
+(o) -i GENERATION_INPUT_SOURCE is the source of input data to use when generating the template - "composite", "mock"
+(o) -p GENERATION_PROVIDER     is an addittional provider which is forced to load from a dir in GENERATION_PLUGIN_DIRS
+
+
+  (m) mandatory, (o) optional, (d) deprecated
+
+DEFAULTS:
+
+
+NOTES:
+
+
+EOF
+}
+
+function main() {
+
+  options "$@" || return $?
+
+  # Defaults
+  LOADER_DIR="${PROVIDER_CACHE_DIR}/_loader"
+  ${GENERATION_DIR}/createTemplate.sh -p "shared" -e loader -o "${LOADER_DIR}" ${TEMPLATE_ARGS} || return $?
+
+  # check for a provider contract
+  if [[ -f "${LOADER_DIR}/loader-providercontract.json" &&  -s "${LOADER_DIR}/loader-providercontract.json" ]]; then
+
+    PROVIDER_CONTRACT="$( cat "${LOADER_DIR}/loader-providercontract.json" | jq '.' || return $? )"
+
+    # Find all steps for install_provider
+    readarray -t provider_sources_json <<< "$( echo "${PROVIDER_CONTRACT}" | jq -c -r '.Stages[].Steps[] | select(.Type == "install_provider" )')"
+
+    provider_state='{}'
+
+    # Add the shared provider state since that is fixed
+    if git -C "${GENERATION_ENGINE_DIR}" rev-parse --is-inside-work-tree &>/dev/null; then
+      engine_ref="$(git -C "${GENERATION_ENGINE_DIR}" rev-parse --abbrev-ref HEAD)"
+      engine_hash="$(git -C "${GENERATION_ENGINE_DIR}" show-ref --heads --tags --hash "${engine_ref}" )"
+      engine_short_hash="$(git -C "${GENERATION_ENGINE_DIR}" rev-parse --short "${engine_hash}" )"
+
+      engine_remote="$( git -C "${GENERATION_ENGINE_DIR}" rev-parse --symbolic-full-name --abbrev-ref @{u} )"
+      engine_remote_url="$( git -C "${GENERATION_ENGINE_DIR}" remote get-url "${engine_remote%%"/${engine_ref}"}" )"
+
+      provider_state="$( update_provider_state "${provider_state}" "_engine" "git" "${engine_short_hash}:${engine_remote_url}" "${GENERATION_ENGINE_DIR}" )"
+
+    else
+      provider_state="$( update_provider_state "${provider_state}" "_engine" "local" "${GENERATION_ENGINE_DIR}" "${GENERATION_ENGINE_DIR}" )"
+    fi
+
+    info "Loading providers from contract..."
+
+    for provider_source in "${provider_sources_json[@]}"; do
+      provider_instance_id="$( echo "${provider_source}" | jq -c -r '.Id' )"
+      provider_name="$( echo "${provider_source}" | jq -c -r '.Parameters.ProviderName' )"
+      source_type="$( echo "${provider_source}" | jq -c -r '.Parameters.Source')"
+      priority="$( echo "${provider_source}" | jq -c -r '.Parameters.Priority')"=
```

Comment:
Is the `=` at the end of the line a typo?

Reply:
Fixed